packages/component-base/test/virtualizer-unlimited-size.test.js

```diff
@@ -202,4 +206,75 @@ describe('unlimited size', () => {
       Math.ceil(itemRect.bottom),
     );
   });
+
+  it('should add more items on size increase', () => {
+    const index = virtualizer.size - 1;
+    virtualizer.scrollToIndex(index);
+    expect(getLastRenderedIndex()).to.equal(index);
+
+    virtualizer.size += 2000;
+    expect(getLastRenderedIndex()).to.be.above(index);
+  });
+
+  it('should remove exceeding items on size decrease', () => {
+    const index = virtualizer.size - 1;
+    virtualizer.scrollToIndex(index);
+    expect(getLastRenderedIndex()).to.equal(index);
+
+    virtualizer.size -= 1000;
+    expect(getLastRenderedIndex()).to.be.below(index);
+  });
+
+  it('should set scroll to end when size decrease affects a visible index', async () => {
+    virtualizer.scrollToIndex(virtualizer.size - 1000);
+    virtualizer.size = virtualizer.firstVisibleIndex - 20;
+    await oneEvent(scrollTarget, 'scroll');
+    const lastItem = elementsContainer.querySelector(`#item-${virtualizer.size - 1}`);
+    expect(lastItem.getBoundingClientRect().bottom).to.be.closeTo(scrollTarget.getBoundingClientRect().bottom, 1);
+  });
+
+  it('should preserve scroll position when size decrease affects a buffered index', () => {
+    // Make sure there are at least 2 buffered items at the end.
+    expect(getLastRenderedIndex() - virtualizer.lastVisibleIndex).to.be.greaterThanOrEqual(2);
+
+    // Scroll to an index and add an additional scroll offset.
+    const index = virtualizer.size - 1000;
+    virtualizer.scrollToIndex(index);
+    scrollTarget.scrollTop += 10;
+
+    // Decrease the size so that the last buffered index exceeds the new size bounds.
+    virtualizer.size = virtualizer.lastVisibleIndex + 1;
+    // await oneEvent(scrollTarget, 'scroll');
```

Comment:
It was commented out as an attempt to fix the CI test failures (not reproducible locally). However, it turned out that the failures were related to the scroll target height – CI fails with 250px, but passes with 200px. So, this line can be uncommented. It would be good to investigate these failures a bit more. Maybe a more reliable solution would be to set a fixed height for items to avoid font rendering differences.